source/slang/slang-lower-to-ir.cpp

```diff
@@ -7294,6 +7289,24 @@ RefPtr<IRModule> TargetProgram::createIRModuleForLayout(DiagnosticSink* sink)
         builder->addLayoutDecoration(irType, irTypeLayout);
     }
 
+    // Lets strip and run DCE here
+    if (linkage->m_obfuscateCode)
+    {
+        IRStripOptions stripOptions;
+
+        stripOptions.shouldStripNameHints = linkage->m_obfuscateCode;
+        stripOptions.stripSourceLocs = linkage->m_obfuscateCode;
+
+        stripFrontEndOnlyInstructions(irModule, stripOptions);
+
+        IRDeadCodeEliminationOptions options;
+        options.keepExportsAlive = true;
+        options.keepLayoutsAlive = true;
+
+        // Eliminate any dead code
+        eliminateDeadCode(irModule, options);
+    }
```

Comment:
It is a small point, but I think this stretch of code is mostly copy-pasted from the other site where we do obfuscation for the "main" module code. It might be good to move this out into a static function at some point since we might need to invoke it from a third place eventually (e.g., I know that we also generate IR code for specalizations, and that would probably have the same need for obfuscation).